src/v/datalake/translation_task.cc

```diff
@@ -36,7 +36,7 @@ ss::future<checked<remote_path, translation_task::errc>> execute_single_upload(
   retry_chain_node& parent_rcn,
   lazy_abort_source& lazy_as) {
     auto file_remote_path = remote_path{
-      file.table_location / remote_path_prefix
+      file.table_location / remote_path_prefix / file.partition_key_path
       / file.local_file.path().filename()};
```

Comment:
Probably unrelated to this PR, but I see that the parquet files are generated with a weird name starting with `-` (e.g. `-7c0a39de-41d7-4531-80d0-ebfb40e054c3.parquet`).

Reply:
This was "broken" in https://github.com/redpanda-data/redpanda/pull/25152/files#diff-617058a8220c70fdc909b169e594470f7b81ed70497f71c95f421c9da374aee8L290

vs 

https://github.com/redpanda-data/redpanda/blob/502834a7ee95102e1280a486b2c386a9656a08cd/src/v/datalake/translation/deps.cc#L482

Reply:
Per an offline discussion with Michal, he'll fix this (and add term/partition-id to the filename) in a followup.